compiler/test/compilable/test23013.d

```diff
@@ -0,0 +1,43 @@
+// https://issues.dlang.org/show_bug.cgi?id=24013
+
+struct PropDescriptor(T)
+{
+    void define(T delegate() aGetter, string aName) {}
+}
+
+enum Get;
+
+mixin template PropertyPublisherImpl()
+{
+    void collectPublicationsFromPairs(T)()
+    {
+        foreach (member; __traits(derivedMembers, T))
+        foreach (overload; __traits(getOverloads, T, member))
+        {
+            alias Attributes = __traits(getAttributes, overload);
+            static if (Attributes.length != 0)
+            {
+                auto descriptor = new PropDescriptor!ulong;
+                auto dg = &overload;
+                descriptor.define(dg, member);
```

Comment:
maybe add a supplemental `descriptor.define(&overload, member);` after this line. As mentioned in the issue, the workaround works, I want to see it still does after your fix

Reply:
With this fix it becomes an error, as it was before. I think this should be filed as a separate issue.